src/3d/qgsrubberband3d.cpp

```diff
@@ -88,58 +91,93 @@ QgsRubberBand3D::QgsRubberBand3D( Qgs3DMapSettings &map, QgsWindow3DEngine *engi
   mMarkerGeometryRenderer->setGeometry( mMarkerGeometry );
   mMarkerGeometryRenderer->setVertexCount( mMarkerGeometry->count() );
 
-  const QVariantMap props
-  {
-    {QStringLiteral( "color" ), QStringLiteral( "red" ) },
-    {QStringLiteral( "size" ), 6 },
-    {QStringLiteral( "outline_color" ), QStringLiteral( "green" ) },
-    {QStringLiteral( "outline_width" ), 0.5 }
-  };
-
-  mMarkerSymbol = QgsMarkerSymbol::createSimple( props );
-  updateMarkerMaterial();
+  setMarkerType( mMarkerType );
   mMarkerEntity->addComponent( mMarkerGeometryRenderer );
 }
 
 QgsRubberBand3D::~QgsRubberBand3D()
 {
-  delete mLineEntity;
-  delete mMarkerEntity;
+  if ( mLineEntity )
+    mLineEntity->deleteLater();
+  mMarkerEntity->deleteLater();
   delete mMarkerSymbol;
 }
 
 float QgsRubberBand3D::width() const
 {
-  return mLineMaterial->lineWidth();
+  return mWidth;
 }
 
 void QgsRubberBand3D::setWidth( float width )
 {
-  mLineMaterial->setLineWidth( width );
+  mWidth = width;
+
+  if ( mGeometryType == Qgis::GeometryType::Line )
+  {
+    // when highlighting lines, the vertex markers should be wider
+    mLineMaterial->setLineWidth( width );
+    width *= 3;
+  }
+
   mMarkerSymbol->setSize( width );
   updateMarkerMaterial();
 }
 
 QColor QgsRubberBand3D::color() const
 {
-  return mLineMaterial->lineColor();
+  return mColor;
 }
 
 void QgsRubberBand3D::setColor( QColor color )
 {
-  mLineMaterial->setLineColor( color );
-  mMarkerSymbol->setColor( color.lighter( 130 ) );
+  mColor = color;
+
+  if ( mGeometryType == Qgis::GeometryType::Line )
+  {
+    mLineMaterial->setLineColor( color );
+    mMarkerSymbol->setColor( color.lighter( 130 ) );
+  }
+  else
+  {
+    mMarkerSymbol->setColor( color );
+  }
+
   if ( mMarkerSymbol->symbolLayerCount() > 0 && mMarkerSymbol->symbolLayer( 0 )->layerType() == QLatin1String( "SimpleMarker" ) )
   {
     static_cast<QgsMarkerSymbolLayer *>( mMarkerSymbol->symbolLayer( 0 ) )->setStrokeColor( color );
   }
   updateMarkerMaterial();
 }
 
+void QgsRubberBand3D::setMarkerType( MarkerType marker )
+{
+  if ( mMarkerSymbol )
+    delete mMarkerSymbol;
```

Comment:
a nitpick - no need to test `if ( mMarkerSymbol )` - we can just delete it even when null. I would also move the deletion just above the creation of new marker symbol. (Or we could even switch to unique_ptr for less hassle?)

Reply:
switched to unique pointer